pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -3086,11 +3134,25 @@ _ReadPolyMesh(_PrimReaderContext* context)
         SdfValueTypeNames->IntArray,
         _CopyGeneric<IInt32ArrayProperty, int>(
             context->ExtractSchema(".faceCounts")));
-    context->AddProperty(
-        _GetUVPropertyName(),
-        _GetUVTypeName(),
-        _CopyGeneric<IV2fGeomParam, GfVec2f>(
-            context->ExtractSchema("uv")));
+
+    // special case for texture coordinates: we want to preserve the topology,
+    // so we need to extract the indices properties if they exist.
+    auto uvProperty = context->ExtractSchema("uv");
+    if (uvProperty.Cast<IV2fGeomParam>().isIndexed()) {
+        context->AddProperty(
+            _GetUVPropertyName(),
+            _GetUVTypeName(),
+            _CopyGeneric<IV2fGeomParam, GfVec2f, false>(uvProperty));
+        context->AddProperty(
+            TfToken(_GetUVPropertyName().GetString() + ":indices"),
+            SdfValueTypeNames->IntArray,
+            _CopyIndices<IV2fGeomParam>(uvProperty));
+    } else {
+        context->AddProperty(
+            _GetUVPropertyName(),
+            _GetUVTypeName(),
+            _CopyGeneric<IV2fGeomParam, GfVec2f>(uvProperty));
+    }
```

Comment:
This is a nice chunk of code that:
a) is duplicated below
b) since UsdGeom and AbcGeom both support indexing for all primvars, we might want to apply to other primvars in the future...

Can you factor it out into a templated helper function, please?